requirements/adjust_versions.py

```diff
@@ -33,28 +32,59 @@ def find_latest(ver: str) -> Dict[str, str]:
     raise ValueError(f"Missing {ver} in {VERSIONS}")
 
 
-def main(path_req: str, torch_version: Optional[str] = None) -> None:
+def main(req: str, torch_version: Optional[str] = None) -> str:
     if not torch_version:
         import torch
 
         torch_version = torch.__version__
     assert torch_version, f"invalid torch: {torch_version}"
 
-    with open(path_req) as fp:
-        req = fp.read()
-    # remove comments
-    req = re.sub(rf"\s*#.*{os.linesep}", os.linesep, req)
+    # remove comments and strip whitespace
+    req = re.sub(rf"\s*#.*{os.linesep}", os.linesep, req).strip()
 
     latest = find_latest(torch_version)
     for lib, version in latest.items():
-        replace = f"{lib}=={version}" if version else lib
-        replace += os.linesep
-        req = re.sub(rf"{lib}[>=]*[\d\.]*{os.linesep}", replace, req)
+        replace = f"{lib}=={version}" if version else ""
+        req = re.sub(rf"\b{lib}(?!\w).*", replace, req)
 
-    print(req)  # on purpose - to debug
-    with open(path_req, "w") as fp:
-        fp.write(req)
+    return req
+
+
+def test():
```

Comment:
Not sure if this test should be there.

Reply:
I couldn't think of any other way to test this, as this is outside of our CI.

It should be very quick to run